src/Microsoft.TestPlatform.Utilities/CodeCoverageRunSettingsProcessor.cs

```diff
@@ -0,0 +1,484 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.VisualStudio.TestPlatform.Utilities
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Text;
+    using System.Xml;
+    using System.Xml.XPath;
+
+    using Microsoft.VisualStudio.TestPlatform.ObjectModel;
+
+    /// <summary>
+    /// Represents the run settings processor for code coverage data collectors.
+    /// </summary>
+    public class CodeCoverageRunSettingsProcessor
+    {
+        #region Type Members
+        /// <summary>
+        /// Represents the exclusion type for code coverage run settings.
+        /// </summary>
+        private class Exclusion
+        {
+            /// <summary>
+            /// Represents the <see cref="XPathNavigator"/> style path of the exclusion type.
+            /// </summary>
+            private string path;
+
+            /// <summary>
+            /// Gets the path components for the current exclusion type.
+            /// </summary>
+            public IEnumerable<string> PathComponents { get; }
+
+            /// <summary>
+            /// Gets the exclusion rules for the current exclusion type.
+            /// </summary>
+            public IDictionary<string, XmlNode> ExclusionRules { get; }
+
+            /// <summary>
+            /// Gets the actual exclusion type path generated from the individual path components.
+            /// </summary>
+            public string Path
+            {
+                get
+                {
+                    if (string.IsNullOrEmpty(this.path))
+                    {
+                        this.path = Exclusion.BuildPath(this.PathComponents);
+                    }
+
+                    return this.path;
+                }
+            }
+
+            /// <summary>
+            /// Constructs an <see cref="Exclusion"/> object.
+            /// </summary>
+            /// 
+            /// <param name="pathComponents">The path split in components.</param>
+            public Exclusion(IEnumerable<string> pathComponents)
+            {
+                this.path = string.Empty;
+                this.PathComponents = pathComponents;
+                this.ExclusionRules = new Dictionary<string, XmlNode>();
+            }
+
+            /// <summary>
+            /// Assembles a relative path from the path given as components.
+            /// </summary>
+            /// 
+            /// <returns>A relative path built from path components.</returns>
+            public static string BuildPath(IEnumerable<string> pathComponents)
+            {
+                var sb = new StringBuilder();
+                sb.Append(".");
+
```

Comment:
the whole method can be done in 1 liner like this

String.Join("/", new [] {"."}.Concat(pathComponents))

Reply:
Done.